addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -306,60 +312,159 @@ private synchronized void startAutomaticRefresh() {
             pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
         }
-        if (sendCommandJob == null || sendCommandJob.isCancelled()) {
-            sendCommandJob = scheduler.scheduleWithFixedDelay(sendCommandRunnable, 0, sendCommandInterval,
-                    TimeUnit.SECONDS);
+        if (queueConsumerThread == null || !queueConsumerThread.isAlive()) {
+            queueConsumerThread = new Thread(new QueueConsumer(commandQueue), "max-queue-consumer");
+            queueConsumerThread.setDaemon(true);
+            queueConsumerThread.start();
         }
     }
 
     /**
      * stops the refreshing jobs
+     *
+     * @throws InterruptedException
      */
-    private void stopAutomaticRefresh() {
+    private void stopAutomaticRefresh() throws InterruptedException {
         if (pollingJob != null && !pollingJob.isCancelled()) {
             pollingJob.cancel(true);
             pollingJob = null;
         }
-        if (sendCommandJob != null && !sendCommandJob.isCancelled()) {
-            sendCommandJob.cancel(true);
-            sendCommandJob = null;
+
+        if (queueConsumerThread != null && queueConsumerThread.isAlive()) {
+            queueConsumerThread.interrupt();
+            queueConsumerThread.join(1000);
+        }
+    }
+
+    public class QueueConsumer implements Runnable {
+        private final BlockingQueue<SendCommand> commandQueue;
+
+        public QueueConsumer(final BlockingQueue<SendCommand> commandQueue) {
+            this.commandQueue = commandQueue;
+        }
+
+        /**
+         * Keeps taking commands from the command queue and send it to
+         * {@link sendCubeCommand} for execution.
+         */
+        @Override
+        public void run() {
+            try {
+                while (!Thread.currentThread().isInterrupted()) {
+                    waitForNormalDutyCycle();
+                    final SendCommand sendCommand = commandQueue.take();
+                    CubeCommand cmd = sendCommand.getCubeCommand();
+                    if (cmd == null) {
+                        cmd = getCommand(sendCommand);
+                    }
+                    if (cmd != null) {
+                        // Actual sending of the data to the Max! Cube Lan Gateway
+                        logger.debug("Command {} sent to MAX! Cube at IP: {}", sendCommand, ipAddress);
+
+                        if (sendCubeCommand(cmd)) {
+                            logger.trace("Command {} completed for MAX! Cube at IP: {}", sendCommand, ipAddress);
+                        } else {
+                            logger.warn("Error sending command {} to MAX! Cube at IP: {}", sendCommand, ipAddress);
+                        }
+                    }
+                    Thread.sleep(50);
+                }
+            } catch (InterruptedException e) {
+                logger.info("Stopping queueConsumer");
```

Comment:
This is probably not an issue here, since it's the last statement of the thread.